autopush/endpoint.py

```diff
@@ -755,12 +753,11 @@ def delete(self, router_type="", router_token="", uaid="", chid=""):
             return self._write_response(
                 400, 108, message="Invalid router")
         router = self.ap_settings.routers[router_type]
-        valid, router_token = router.check_token(router_token)
+        relo_url = self._relocate(router_type, uaid, chid)
+        valid, reply = router.check_token(router_token, self._write_response,
+                                          relo_url)
         if not valid:
-            newUrl = self._relocate(router_type, router_token, uaid, chid)
-            return self._write_response(
-                301, 0, "Location: %s" % newUrl,
-                headers={"Location": newUrl})
+            return reply
```

Comment:
This is always a None value, since only the GCM check_token even calls self._write_response, and the return value of self._write_response is None.